### PR TITLE
Integration testing frontend via Cypress

### PR DESCRIPTION
**Pull request by alexjmacdonald**
### What changes does this PR make to Grapl? Why?

Adds dockerized Cypress as an integration test, with demo testing stubs.  This adds stubs for browser-based tests for the front-end.

Adds `run-engagement-view-integration-tests` and `js-integration-tests` as dobi jobs.

### How were these changes tested?

Should not have any impact on production; they are themselves tests.  Running the commands shows Cypress tests executing in the console.

### CODE REVIEW

File: test/cypress/fixtures/unique_cmd_parent/main.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
+from typing import Any, Type
```
Comment (wimax-grapl) [inline test/cypress/fixtures/unique_cmd_parent/main.py:1]: I don't like that these are copy-pasted. 

Instead, you can copy these from the following image, and put them in the right place.
```
COPY --from=grapl/etc-build /home/grapl/etc 
```

File: test/cypress/plugins/index.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+/**
+ * @type {Cypress.PluginConfig}
+ */
+module.exports = (on, config) => {
+  // `on` is used to hook into various events Cypress emits
+  // `config` is the resolved Cypress config
+  on("before:browser:launch", (browser = {}, launchOptions) => {
+    if (browser.name === "chrome") {
+      launchOptions.args.push("--disable-dev-shm-usage");
```
Comment (wimax-grapl) [inline test/cypress/plugins/index.js:23]: can you leave a comment on why this option is here

File: test/cypress/support/commands.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
+// ***********************************************
+// This example commands.js shows you how to
```
Comment (wimax-grapl) [inline test/cypress/support/commands.js:2]: remove leftover verbiage

**Comment by alexjmacdonald** [comment]
Just realized this branch has some working commits from writing tests with Andrea.  I'm gonna close this out, address the comments & push up a new PR.

**Pull request closed without merging**